datafusion/sql/src/select.rs

```diff
@@ -214,7 +215,7 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
 
         let plan = if let Some(having_expr_post_aggr) = having_expr_post_aggr {
             LogicalPlanBuilder::from(plan)
-                .filter(having_expr_post_aggr)?
+                .having(having_expr_post_aggr)?
```

Comment:
It's important to mention the filter is used for the `HAVING` clause because the wildcard expansion behavior is different.